src/redux/epics/upload-epic.js

```diff
@@ -37,6 +43,61 @@ const initializeUpload = (action$, store) => {
   });
 };
 
+const initializeSession = (action$, store) => {
+  return action$.ofType(uploadActions.INITIALIZE_SESSION).mergeMap(action => {
+    const {chunks, fileName, handle} = action.payload;
+
+    return Observable.fromPromise(Backend.initializeUploadSession(chunks, fileName, handle)).map(
+      ({alphaSessionId, betaSessionId, invoice, numberOfChunks, handle, fileName, genesisHash, storageLengthInYears, host}) => {
+          return uploadActions.pollPaymentStatus({
+            host,
+            alphaSessionId,
+            chunks,
+            fileName,
+            handle,
+            numberOfChunks,
+            betaSessionId,
+            genesisHash,
+            invoice
+          });
+        }
+      );
+  });
+};
+
+const pollPaymentStatus = (action$, store) => {
+  return action$.ofType(uploadActions.POLL_PAYMENT_STATUS).mergeMap(action => {
+    const {host, alphaSessionId, chunks, fileName, handle, numberOfChunks, betaSessionId, genesisHash, invoice} = action.payload;
+
+    return Observable.interval(3000)
+      .startWith(0)
```

Comment:
Did you have to do anything special to get startWith to work?  When I tried to use startWith I got errors.

Reply:
no issues, i just had to use it to fire the interval immediately.